pytorch_lightning/trainer/trainer.py

```diff
@@ -475,7 +473,8 @@ def train(self):
 
         try:
             # run all epochs
-            for epoch in range(self.current_epoch, self.max_epochs):
+            epochs = range(self.current_epoch, self.max_epochs) if self.max_epochs else count(self.current_epoch)
+            for epoch in epochs:
```

Comment:
what happens if `self.max_epochs` is `None`, and `self.max_steps` is large enough that we make more than 1 pass through the data?  the `count(self.current_epoch)` would return an iterable over a single value right? does that mean the loop would stop too early?

Reply:
Don't believe so, unless I'm misunderstanding? the `count()` iterator is indefinite/doesn't raise `StopIteration`.